### DISCUSSION TITLE
How to manage database properly?

### DISCUSSION THREAD

**Discussion by Riddle356** [hjlarry/flask-shop#106] [General]
I am new at flask and just started working with docker. Installed everything on my pc and succesfully started application but here is a question. How can I enter mysql database using mysql -u -p command? I entered mysql container than I have tried something like mysql -u root -p 123456 but it denies everytime I try to do it.

**Comment by dedalgr** [comment]
Lock in config for user and passwd

**Reply by Riddle356** [reply]
Yeah, I see db setting in setting.py file and they look like this:

```
class DBConfig:
    db_type = os.getenv("DB_TYPE", "mysql")
    user = os.getenv("DB_USER", "root")
    passwd = os.getenv("DB_PASSWD", "123456")
    host = os.getenv("DB_HOST", "127.0.0.1")
    port = os.getenv("DB_PORT", 3307)
    db_name = os.getenv("DB_NAME", "flaskshop")
    if db_type == "postgresql":
        db_uri = f"postgresql://{user}:{passwd}@{host}:{port}/{db_name}"
    elif db_type == "mysql":
        db_uri = (
            f"mysql+pymysql://{user}:{passwd}@{host}:{port}/{db_name}?charset=utf8mb4"
        )
    redis_uri = "redis://localhost:6379"
    esearch_uri = "localhost"
```

Nevertheless I can't connect to a database through installed mysql on Ubuntu:

mysql -p 3307 -u root -p
Enter password:
ERROR 1698 (28000): Access denied for user 'root'@'localhost'

**Reply by dedalgr** [reply]
os.getenv take defalt if mising config file. Look in config file.
root is not corect user. 
I'm not working on this project seriously. I just added localization and postgresql. Anyway, look in your .env thing

**Comment by hjlarry** [comment]
You can view the `docker-compose.yml` file, the default mysql config is defined here. so the default passwd in docker is `root`.

**Reply by Riddle356** [reply]
Thank you! It works fine
